go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -542,13 +599,19 @@ func (app *registryApplication) registerRuntime( // nolint: gocyclo
 		return err
 	}
 
-	// Make sure the signer of the transaction matches the signer of the runtime.
+	// Make sure the signer of the transaction matches the signer of the
+	// entity that is controlling the runtime.
 	// NOTE: If this is invoked during InitChain then there is no actual transaction
 	//       and thus no transaction signer so we must skip this check.
-	if !ctx.IsInitChain() && !sigRt.Signature.PublicKey.Equal(ctx.TxSigner()) {
+	if !ctx.IsInitChain() && !rt.EntityID.Equal(ctx.TxSigner()) {
```

Comment:
This actually needs to take into account the existing governance model. So this check is true for entity governance, but for runtime governance the check would look something like checking `ctx.CallerAddress()` for equality with `staking.NewRuntimeAddress(rt.ID)`.

So maybe put a switch here that decides based on the governance model. The switch could also incorporate the conditional below which returns `ErrForbidden` in case of consensus governance.

Reply:
Thanks, fixed! :)